paths_cli/wizard/shooting.py

```diff
@@ -0,0 +1,128 @@
+from functools import partial
+
+from paths_cli.wizard.core import get_missing_object
+from paths_cli.wizard.engines import engines
+from paths_cli.wizard.cvs import cvs
+from paths_cli.parsing.tools import custom_eval
+
+import numpy as np
+
+
+def uniform_selector(wizard):
+    import openpathsampling as paths
+    return paths.UniformSelector()
+
+def gaussian_selector(wizard):
+    import openpathsampling as paths
+    cv_name = wizard.ask_enumerate("Which CV do you want the Gaussian to "
+                                   "be based on?",
+                                   options=wizard.cvs.keys())
+    cv = wizard.cvs[cv_name]
+    l_0 = wizard.ask_custom_eval(f"At what value of {cv.name} should the "
+                                 "Gaussian be centered?")
+    std = wizard.ask_custom_eval("What should be the standard deviation of "
+                                 "the Gaussian?")
+    alpha = 0.5 / (std**2)
+    selector = paths.GaussianBiasSelector(cv, alpha=alpha, l_0=l_0)
+    return selector
+
+# def random_velocities(wizard):
+    # pass
+
+# def gaussian_momentum_shift(wizard):
+    # pass
+
+# def get_allowed_modifiers(engine):
+    # allowed = []
+    # randomize_attribs = ['randomize_velocities', 'apply_constraints']
+    # if any(hasattr(engine, attr) for attr in randomize_attribs):
+        # allowed.append('random_velocities')
+
+    # if not engine.has_constraints():
+        # allowed.append('velocity_changers')
+
+    # return allowed
+
+SHOOTING_SELECTORS = {
+    'Uniform random': uniform_selector,
+    'Gaussian bias': gaussian_selector,
+}
+
+
+def _get_selector(wizard, selectors=None):
+    if selectors is None:
+        selectors = SHOOTING_SELECTORS
+    selector = None
+    sel = wizard.ask_enumerate("How do you want to select shooting "
+                               "points?", options=list(selectors.keys()))
+    selector = selectors[sel](wizard)
+    return selector
+
+def one_way_shooting(wizard, selectors=None, engine=None):
+    from openpathsampling import strategies
+    if engine is None:
+        engine = get_missing_object(wizard, wizard.engines, 'engine',
+                                    engines)
+
+    selector = _get_selector(wizard, selectors)
+    strat = strategies.OneWayShootingStrategy(selector=selector,
+                                              engine=engine)
+    return strat
+
+# def two_way_shooting(wizard, selectors=None, modifiers=None):
+    # pass
+
+def spring_shooting(wizard, engine=None):
+    import openpathsampling as paths
```

Comment:
any reason why these are not module level imports?

Reply:
Yes. Module level imports would be called every time the CLI runs, and the `openpathsampling` import is slow (I profiled this once, and IIRC the issue is in netcdf -- so we might be able to fix this for 2.0). Slow imports would mean:

* slow `openpathsampling -h`
* slow autocompletion of subcommands

I don't currently use the subcommand autocompletion because there was a bug in click 7.x where subcommand completion for zsh disabled completion of filenames in the same command (and I'd rather have filename completion than subcommand completion). However, I think this was fixed in click 8.0. I haven't tried it out yet, but shell completion is easy to enable on the user side: https://click.palletsprojects.com/en/8.0.x/shell-completion/